Presentation.Web/Swagger/FilterByApiVersionFilter.cs

```diff
@@ -19,12 +20,32 @@ public FilterByApiVersionFilter(Func<SwaggerDocument, int> getApiVersion, Func<s
         public void Apply(SwaggerDocument swaggerDoc, SchemaRegistry schemaRegistry, IApiExplorer apiExplorer)
         {
             var docVersion = _getApiVersion(swaggerDoc);
-            foreach (var path in swaggerDoc.paths.ToList())
+            foreach (var path in swaggerDoc.paths.Where(path => docVersion != _getPathApiVersion(path.Key)).ToList())
             {
-                if (docVersion != _getPathApiVersion(path.Key))
-                {
-                    swaggerDoc.paths.Remove(path);
-                }
+                swaggerDoc.paths.Remove(path);
+            }
+
+            var listOfSchemaWithReference = swaggerDoc.paths
```

Comment:
See this:
![image](https://user-images.githubusercontent.com/13116889/216819852-40a3be3c-417d-4e74-a57b-4458a4d09a09.png)

More filtering actually goes on so for the filtering to apply correctly, this code (the type filtering) should be placed in it's own documentfilter `PurgeUnusedTypesDocumentFilter` and added to the pipeline here:
![image](https://user-images.githubusercontent.com/13116889/216819992-d1e047ec-7738-4dc8-a006-beaac7e4ae8b.png)